src/spooled.rs

```diff
@@ -157,7 +157,11 @@ impl Write for SpooledTempFile {
         // roll over to file if necessary
         if matches! {
             &self.inner, SpooledData::InMemory(cursor)
-            if cursor.position() as usize + buf.len() > self.max_size
+            if cursor
+                .position()
+                .checked_add(buf.len() as u64)
+                .filter(|&val| val <= self.max_size as u64)
+                .is_none()
```

Comment:
You could seek up to usize::MAX before calling write, then it will overflow. I haven't added a test case for this, because it could lead to a flush.

Reply:
In fact, the test is easy. Added to commit for reference.

Reply:
Can we rewrite it with saturating add? I.e.: `cursor.position.saturating_add(buf.len() as u64) > self.max_size as u64`.

Reply:
Yes, fortunately we can. I've had the false assumption that it would be an issue if `self.max_size` itself is `usize::MAX` but in this case, the underyling `write` calls will fail anyway.

Thanks for feedback so these `is_none`-checks can be removed from both places!